modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -945,6 +946,121 @@ public String getTypeDeclaration(Schema p) {
         return super.getTypeDeclaration(target);
     }
 
+    private String getBeanValidation(Schema<?> items) {
+        if (Boolean.FALSE.equals(additionalProperties.getOrDefault(BeanValidationFeatures.USE_BEANVALIDATION, Boolean.FALSE))) {
+            return "";
+        }
+
+        if (ModelUtils.isTypeObjectSchema(items)) {
+            // prevents generation '@Valid' for Object
+            return "";
+        }
+
+        if (items.get$ref() != null) {
+            return "@Valid ";
+        }
+
+        if (ModelUtils.isStringSchema(items)) {
+            return getStringBeanValidation(items);
+        }
+
+        if (ModelUtils.isNumberSchema(items)) {
+            return getNumberBeanValidation(items);
+        }
+
+        if (ModelUtils.isIntegerSchema(items)) {
+            return getIntegerBeanValidation(items);
+        }
+
+        return "";
+    }
+
+    private String getIntegerBeanValidation(Schema<?> items) {
+        if (items.getMinimum() != null && items.getMaximum() != null) {
+            return String.format(Locale.ROOT, "@Min(%s) @Max(%s)", items.getMinimum(), items.getMaximum());
```

Comment:
This is generating code - I don't think it should be taking in a locale, since it should be the same regardless of the machine/user Locale settings

Reply:
Probably better for @wing328 or someone with broader context to comment it just seems odd to use a Locale to format strings which are supposed to represent code.

If that's the standard for the project, probably best to leave as-is.

Reply:
@wing328, sorry to bother you, have you had a chance to look into  it? do I need to make additional changes or can I resolve this conversation ?

Reply:
let me take another look tomorrow.

Reply:
@wing328 hey hey! sorry to bother you again... any updates?

note: I noticed that the tests for some samples in the extended test checks did not pass due to the lack of imports .. that’s why I added imports to the templates .. I hope I found all the places where this is needed, let me know if it’s ok

Reply:
checking now

Reply:
some java sample tests still failed. can you please take a look when you've time?

Reply:
oh sorry for that,  I missed that for helidon should use  rootJavaEEPackage in templates, fixed  hope now all tests will passed. Thanks!